packages/@lwc/template-compiler/src/shared/ir.ts

```diff
@@ -93,7 +91,7 @@ export function isComponentProp(identifier: TemplateIdentifier, root: IRNode): b
             }
         }
 
-        current = current.parent;
+        current = size > 0 ? parentStack[--size] : undefined;
```

Comment:
Instead of using a while loop, you can use a simple for:

```js
for (let i = parentStack.length - 1; i >= 0; i--) {
  current = parentStack[i];
  // [...]
} 
```

Reply:
I updated isInIteration from parser/index.ts and isComponentProp to use the for loop.